### PR TITLE
LPS-197432 Fix indirect and direct related properties in API Builder

### PR DESCRIPTION
**Pull request by sergiojimcos**
## Aim of this PR
Changes from this [PR](https://github.com/liferay-headless/liferay-portal/pull/1550)

The aim of this PR is to solve direct and indirect API Properties from different Object Definitions. Apparently, it is not possible to create Object Entries with null object fields because it throws an Exception. However, it is possible to return API Properties where hasn't any related API Properties and this is retrieving in a empty array, this use case is control.

**Changes**
- Changes in the EndpointHelper to retrieve not null related object values

**Steps to reproduce**
1. Go to this [JIRA Ticket ](https://liferay.atlassian.net/browse/LPS-197432)and reproduce all the steps.

**Related JIRA Ticket**
https://liferay.atlassian.net/browse/LPS-197432

### CODE REVIEW

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
@jalber786 can you please add an integration test that does this?

```bash
curl -X "DELETE" -u 'test@liferay.com:test' "http://localhost:8080/o/headless-builder/applications/by-external-reference-code/LPS-197432"
curl -X 'POST' \
  'http://localhost:8080/o/headless-builder/applications/' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -u 'test@liferay.com:test' \
  -d '{
  "externalReferenceCode": "LPS-197432",
  "applicationStatus": "published",
  "baseURL": "lps-197432",
  "description": "Reproduces LPS-197432.",
  "title": "LPS-197432"
}'

curl -X 'POST' \
  'http://localhost:8080/o/headless-builder/endpoints/' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -u 'test@liferay.com:test' \
  -d '{
  "externalReferenceCode": "LPS-197432-endpoint",
  "description": "This endpoint returns the names of all the existing endpoints and their schemas",
  "httpMethod": "get",
  "name": "getEndpointSchema",
  "path": "/getendpointschema",
  "r_apiApplicationToAPIEndpoints_c_apiApplicationERC": "LPS-197432",
  "responseAPISchemaToAPIEndpoints": {
    "externalReferenceCode": "LPS-197432-schema",
    "description": "The endpoint schema",
    "mainObjectDefinitionERC": "L_API_ENDPOINT",
    "name": "Endpoint",
    "r_apiApplicationToAPISchemas_c_apiApplicationERC": "LPS-197432",
    "apiSchemaToAPIProperties": [
      {
        "externalReferenceCode": "endpoint-name-property-erc",
        "description": "The name of the endpoint",
        "name": "endpointName",
        "objectFieldERC": "NAME"
      },
      {
        "externalReferenceCode": "schema-name-property-erc",
        "description": "The name of the schema",
        "name": "schemaName",
        "objectFieldERC": "NAME",
        "objectRelationshipNames": "apiApplicationToAPIEndpoints,apiApplicationToAPISchemas"
      }
    ]
  },
  "retrieveType": "collection",
  "scope": "company"
}'
curl -X 'GET' -u 'test@liferay.com:test' 'http://localhost:8080/o/c/lps-197432/getendpointschema'
```

**Comment by jalber786** [comment]
ci:test:sf

**Comment by jalber786** [comment]
ci:test:relevant

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/resource/test/HeadlessBuilderResourceTest.java [hunk 1]
```diff
@@ -929,6 +929,79 @@ public void testGetWithCompanyScopedEndpoint() throws Exception {
 			HTTPTestUtil.invokeToHttpCode(null, endpoint2, Http.Method.GET));
 	}
 
+	@Test
+	public void testGetWithIndirectlyRelatedProperty() throws Exception {
```
Comment (4lejandrito) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/resource/test/HeadlessBuilderResourceTest.java:933]: Looks good!

**Review by 4lejandrito — APPROVED** [review]

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by 4lejandrito** [comment]
ci:forward

**Review by jalber786 — APPROVED** [review]

**Comment by sergiojimcos** [comment]
ci:forward:force

**Pull request closed without merging**